### PR TITLE
Fix nested sub-options rendering

### PR DESCRIPTION
**Pull request by IlyaKhD**

### CODE REVIEW

File: src/core/__tests__/nested-option.test.tsx [hunk 1]
```diff
@@ -19,124 +29,287 @@ class CollectionNestedComponent extends ConfigurationComponent<{ c?: number, d?:
...
-
-    expect(WidgetClass.mock.calls[0][1]).toEqual({
-        templatesRenderAsynchronously: true,
-        itemOptions: [
-            { c: 123, d: "abc" }
-        ]
+    it("is pulled overriden if not a collection item", () => {
+        mount(
+            <TestComponent>
+                <NestedComponent a={123} />
```
Comment (lukyanovas) [inline src/core/__tests__/nested-option.test.tsx]: Maybe, in this case, we should also throw an exception.

File: src/core/__tests__/props-updating.test.tsx [hunk 2]
```diff
@@ -208,112 +246,83 @@ describe("controlled mode", () => {
...
+            </ControlledComponent>
 
-        it("ignores 3rd-party changes in nested default props if parent object changes", () => {
-            mount(
-                <ControlledComponent>
-                    <NestedComponent defaultC="default" />
-                </ControlledComponent>
-            );
+        );
+        const nested = component.find(NestedComponent).dive();
```
Comment (lukyanovas) [inline src/core/__tests__/props-updating.test.tsx]: I prefer using `mount` in such cases

File: src/core/__tests__/nested-option.test.tsx (hunk 1 above)
Comment (IlyaKhD) [reply src/core/__tests__/nested-option.test.tsx]: Added this to our inbox.

File: src/core/__tests__/props-updating.test.tsx (hunk 2 above)
Comment (IlyaKhD) [reply src/core/__tests__/props-updating.test.tsx]: In these tests `.shallow()` renders the widget component and `.dive()` renders the nested option component - that's enough (all the components are rendered). 

Moreover, using `.mount()` makes impossible to set options for non-root components:
```console
ReactWrapper::setProps() can only be called on the root
```
See https://github.com/airbnb/enzyme/issues/1289.

**Pull request merged**